### PR TITLE
Add benchmarks to nonlinear BVPs

### PR DESCRIPTION
**Pull request by michelle-zong**

### CODE REVIEW

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 1]
```diff
@@ -0,0 +1,213 @@
+---
+title: Nonlinear BVP Benchmarks
+author: Michelle Zong
+TESTING
```
Comment (avik-pal) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:4]: Remove this. This is not valid YAML Syntax

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+### Nonlinear BVP 1
+
+```julia
+prob_1 = BVProblemLibrary.prob_bvp_nonlinear_1
+sol_1 = solve(prob_1, MIRK3(), abstol=1e-8, reltol=1e-8, dt = 0.1)
+testsol_1 = TestSolution(sol_1.t, sol_1.u)
+wp_1 = WorkPrecisionSet(prob_1, abstols, reltols, setups; names = labels, print_names = true, appxsol = testsol_1, 
+                        dense=false, save_everystep=false,numruns=100,maxiters=10000000, timeseries_errors=false,verbose=false)
+plot(wp_1)
```
Comment (avik-pal) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:53]: Reduce the maxiters and numruns for now. We can increase them once we have some results

**Review by avik-pal — CHANGES_REQUESTED** [review]

**Comment by avik-pal** [comment]
@michelle-zong lets add `Shooting` and `MultipleShooting` in here as well. For the internal ODE solver corresponding to the tolerances see https://docs.sciml.ai/DiffEqDocs/dev/solvers/ode_solve/

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 3]
```diff
@@ -0,0 +1,216 @@
...
+            Dict(:alg=>MultipleShooting(10,Tsit5())),
+            Dict(:alg=>MultipleShooting(10,Vern7()))]
+labels = ["MIRK2";
+               "MIRK3";
+               "MIRK4";
+               "MIRK5";
+               "MIRK6";
+               "Shooting(Tsit5)";
+               "Shooting(Vern7)";
+               "MultipleShooting(Tsit5)",
```
Comment (ErikQQY) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:31]:
```suggestion
               "MultipleShooting(Tsit5)";
```

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 4]
```diff
@@ -39,6 +39,24 @@ abstols = 1.0 ./ 10.0 .^ (1:4)
...
+Function that checks retcode.
+
+```julia
+function retcode_check(setups, labels)
+    success_setups = []
+    success_labels = []
+    for (setup, label) in zip(setups, labels)
+        temp_sol = solve(prob, setup[:alg], abstol=1/10^8,reltol=1/10^8, dt=0.1)
+        test = TestSolution(temp_sol.t, temp_sol.u)
+        if SciMLBase.successful_retcode(test.retcode)
```
Comment (avik-pal) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:51]: What is this checking? It should be checking the recode fo the solution returned by `solve`

**Comment by ErikQQY** [comment]
Some rebase would make Buildkite work

**Comment by michelle-zong** [comment]
> Some rebase would make Buildkite work

**Comment by michelle-zong** [comment]
> Some rebase would make Buildkite work

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd (hunk 4 above)
Comment (michelle-zong) [reply benchmarks/NonStiffBVP/nonlinear_wpi.jmd]: Should be fixed now

File: benchmarks/NonStiffBVP/Project.toml [hunk 5]
```diff
@@ -1,4 +1,5 @@
 [deps]
+BVProblemLibrary = "ded0fc24-dfea-4565-b1d9-79c027d14d84"
```
Comment (ErikQQY) [inline benchmarks/NonStiffBVP/Project.toml:2]: These changes in Project.toml and Manifest.toml are unnecessary.

**Review by ErikQQY — CHANGES_REQUESTED** [review]

File: benchmarks/NonStiffBVP/Project.toml (hunk 5 above)
Comment (avik-pal) [reply benchmarks/NonStiffBVP/Project.toml:2]: I think BVProblemLibrary is needed, right? But yeah `Weave` should not be added.

Comment (avik-pal) [reply benchmarks/NonStiffBVP/Project.toml:2]: Oh wait this is duplicated. Erik is right, this needs to be removed

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 6]
```diff
@@ -0,0 +1,247 @@
...
+```
+
+Function that checks retcode.
+
+```julia
+function retcode_check(setups, labels)
+    success_setups = []
+    success_labels = []
+    for (setup, label) in zip(setups, labels)
+        temp_sol = solve(prob, setup[:alg], abstol=1/10^8,reltol=1/10^8, dt=0.1)
```
Comment (ErikQQY) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:49]: The CI failed because ```prob``` here in this function is not defined

File: benchmarks/NonStiffBVP/nonlinear_wpi.jmd [hunk 7]
```diff
@@ -0,0 +1,247 @@
+
+This is a benchmark of MIRK methods and Shooting method on some nonlinear boundary value problems. The testing BVPs are a set of standard BVP test problems as described [here](https://archimede.uniba.it/~bvpsolvers/testsetbvpsolvers/?page_id=29)
+
+# Setup
+
+Fetch required packages.
+
+```julia
+using BoundaryValueDiffEq, OrdinaryDiffEq, DiffEqDevTools, BenchmarkTools, BVProblemLibrary, Weave, Plots
```
Comment (avik-pal) [inline benchmarks/NonStiffBVP/nonlinear_wpi.jmd:9]:
```suggestion
using BoundaryValueDiffEq, OrdinaryDiffEq, DiffEqDevTools, BenchmarkTools, BVProblemLibrary, Plots
```

**Comment by ChrisRackauckas** [comment]
Superseded by https://github.com/SciML/SciMLBenchmarks.jl/pull/1203
